### PR TITLE
refactor: Remove OCMock Carthage dependency

### PR DESCRIPTION
**Pull request by dplewis**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-iOS-OSX/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-iOS-OSX/issues?q=is%3Aissue).

### Issue Description

https://github.com/parse-community/Parse-SDK-iOS-OSX/issues/1747

Closes: N/A

### Approach

* Update minimum deployment versions.
* Add minimum deployment versions to README
* Add SPM caching to CI
* Expose ParseFacebookUtils
* Install OCMock via SPM
* Clean up missing file references

### TODOs before merging

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)

### CODE REVIEW

**Comment by mtrezza** [comment]
Removed myself from review since there have been more commits after the request. When this is ready for review please request again from `@parse-community/ios-sdk-review`

**Comment by dplewis** [comment]
I would but it looks like you aren't a member of @parse-community/ios-sdk-review. Do you get notified?

**Comment by mtrezza** [comment]
You are right; I'm now

File: Parse/Parse/Resources/Parse-iOS.Info.plist [hunk 1]
```diff
@@ -23,6 +23,6 @@
 	<key>CFBundleVersion</key>
 	<string>2.7.3</string>
 	<key>MinimumOSVersion</key>
-	<string>9.0</string>
+	<string>12.0</string>
```
Comment (mtrezza) [inline Parse/Parse/Resources/Parse-iOS.Info.plist:26]: Is this bumping the required min iOS  version of the SDK, ergo a breaking change?

File: ParseFacebookUtils/ParseFacebookUtils/Resources/Info-iOS.plist [hunk 2]
```diff
@@ -23,6 +23,6 @@
 	<key>CFBundleVersion</key>
 	<string>2.7.3</string>
 	<key>MinimumOSVersion</key>
-	<string>9.0</string>
+	<string>12.0</string>
```
Comment (mtrezza) [inline ParseFacebookUtils/ParseFacebookUtils/Resources/Info-iOS.plist:26]: Same here

File: ParseFacebookUtilsiOS/ParseFacebookUtilsiOS/Resources/Info-iOS.plist [hunk 3]
```diff
@@ -23,6 +23,6 @@
 	<key>CFBundleVersion</key>
 	<string>1.19.4</string>
 	<key>MinimumOSVersion</key>
-	<string>9.0</string>
+	<string>12.0</string>
```
Comment (mtrezza) [inline ParseFacebookUtilsiOS/ParseFacebookUtilsiOS/Resources/Info-iOS.plist:26]: Same here

File: Parse/Parse/Resources/Parse-iOS.Info.plist (hunk 1 above)
Comment (dplewis) [reply Parse/Parse/Resources/Parse-iOS.Info.plist:26]: The minimum version is 12. Check `package.swift`. 9.0 is wrong

**Review by mtrezza — APPROVED** [review]
Looks good; is this ready for merge?

**Comment by dplewis** [comment]
Yes

**Review by mtrezza — COMMENTED** [review]
Actually it seems that this is a breaking change, because it is removing watchOS 2 and 3 compatibility, right?

Package.swift is currently at `.watchOS(.v2)`.

**Comment by dplewis** [comment]
Watch OS 2 and 3 are over 7 years old. Do what you will with that information.

**Comment by mtrezza** [comment]
Is the removal of watchOS 2 and 3 compatibility required to remove the OCMock Carthage dependency?

**Comment by dplewis** [comment]
OCMock minimum version is watchOS 4, I'm correcting it in this PR. 

For reverence

Bolts-Swift is v3
Bolts-Objc is v2
Starscream is v2

**Comment by dplewis** [comment]
@mtrezza If you build this PR you might see `dependency 'OCMock is not used by any target`. It's basically a dev dependency.

WatchOS 4 is the minimum to run Xcode 14+
```
The Xcode 14 release supports on-device debugging in iOS 11 and later, tvOS 11 and later, and watchOS 4 and later. Xcode 14 requires a Mac running macOS Monterey 12.5 or later.
```

**Comment by mtrezza** [comment]
So the watchOS bump is unrelated to the removal of OCMock?

**Comment by dplewis** [comment]
@mtrezza Changes have been made. Flaky test causing CI to fail.

**Comment by mtrezza** [comment]
Do you want to open another PR to bump the deployment target to watchOS 4?

**Comment by dplewis** [comment]
We support Xcode 14+, any developer making a WatchOS app is forced to use WatchOS 4+. 4 > 2  so there will never be an issue.

**Comment by mtrezza** [comment]
But why did you bump this in the first place then?

**Comment by dplewis** [comment]
I thought I was right at the time. I'm still learning. This project hasn't been worked on in a long time. I know why this project supports iOS 12 as a minimum do you?

**Comment by mtrezza** [comment]
I've mentioned my concerns about support policy of older versions in https://github.com/parse-community/Parse-SDK-iOS-OSX/issues/1755. Maybe the discussion should continue there.

**Comment by dplewis** [comment]
Do what you have to, just don't increase the CI build time. Thats the only reason by I'm working on this project right now. I don't work on the parse-server anymore because of the / your `policy`

**Comment by mtrezza** [comment]
Not sure what you mean by that, but as always please feel free to open an issue in the Parse Server repository if you feel that a policy there is making it difficult for you to contribute.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.0.0](https://github.com/parse-community/Parse-SDK-iOS-OSX/releases/tag/3.0.0)

**Pull request merged**